ecom_test/tests/backend_tests/product_reviews/test_products_reviews_statuses.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+from ecom_test.src.generic_helpers.generic_product_helper import GenericProductHelper
+from ecom_test.src.generic_helpers.generic_product_review_helper import GenericProductReviewsHelper
+from ecom_test.src.utilities.generic_utilities import generate_random_string, generate_random_email_and_password
+import logging as logger
+
+
+@pytest.fixture(scope="function")
+def setup_teardown(request):
+    # Create instances of the necessary helper classes
+    request.cls.product_helper = GenericProductHelper()
+    request.cls.product_review_helper = GenericProductReviewsHelper()
+
+    # Create a product and store its ID for reference
+    request.cls.product = request.cls.product_helper.create_a_product()
+    request.cls.rating_count_before = request.cls.product['rating_count']
+    request.cls.product_id = request.cls.product['id']
+
+    yield request.cls.product_helper, request.cls.product_review_helper, request.cls.product_id
```

Comment:
I dont think you have to yield all those objects. you can just say 'yield'. You are setting the variables as request.cls so they are available as self. in the tests that use the fixture.
I think so. Try it and let me know if i am wrong.

Reply:
Ya, this time I wrote differently as I did not use ".self" to call the function in tests. And it is true that I set the variables as request class; however, I did not call any function by calling  .self. But I will take care of this in future